lib/rubocop/cop/naming/method_name.rb

```diff
@@ -31,9 +31,27 @@ module Naming
       class MethodName < Cop
         include ConfigurableNaming
         include IgnoredPattern
+        include RangeHelp
 
         MSG = 'Use %<style>s for method names.'
 
+        def_node_matcher :attr?, <<~PATTERN
```

Comment:
Btw, there's also `define_method` that should be considered. In general it might be a good idea be able to provide a list of method names that define methods via configuration so custom macros could be accounted as well.